pkg/vere/main.c

```diff
@@ -1726,6 +1727,182 @@ _cw_prep(c3_i argc, c3_c* argv[])
+  if ( c3n == u3_lmdb_gulf(old_u->mdb_u, &fir_d, &las_d) ) {
+    fprintf(stderr, "chop: failed to load latest event from database\r\n");
+    u3_king_bail();
+  }
+
+  // get the metadata
+  c3_d     who_d[2];
+  c3_o     fak_o;
+  c3_w     lif_w;
+  if ( c3y != u3_disk_read_meta(old_u->mdb_u, who_d, &fak_o, &lif_w) ) {
+    fprintf(stderr, "chop: failed to read metadata\r\n");
+    u3_king_bail();
+  }
+
+  // get the last event
+  u3_lmdb_walk itr_u;
+  size_t       len_i;
+  void*        buf_v[1];
+  if ( c3n == u3_lmdb_walk_init(old_u->mdb_u, &itr_u, las_d, las_d) ) {
+    fprintf(stderr, "chop: failed to initialize iterator\r\n");
+    u3_king_bail();
+  }
+  if ( c3n == u3_lmdb_walk_next(&itr_u, &len_i, buf_v) ) {
+    fprintf(stderr, "chop: failed to read event\r\n");
+    u3_king_bail();
+  }
+  u3_lmdb_walk_done(&itr_u);
+
+  // initialize a fresh lmdb environment in the "chop" subdir
+  c3_c cho_c[8193];
+  snprintf(cho_c, sizeof(cho_c), "%s/chop", log_c);
+  c3_mkdir(cho_c, 0700);
+  MDB_env* new_u = u3_lmdb_init(cho_c, siz_i);
+
+  // check if the new database was initialized
+  if ( !new_u ) {
+    fprintf(stderr, "chop: failed to initialize new database\r\n");
+    u3_king_bail();
+  }
+
+  // write the metadata to the database
+  if ( c3n == u3_disk_save_meta(new_u, who_d, fak_o, lif_w) ) {
+    fprintf(stderr, "chop: failed to save metadata\r\n");
+    u3_king_bail();
+  }
+
+  // write the last event to the database
+  // warning: this relies on the old database still being open
+  if ( c3n == u3_lmdb_save(new_u, las_d, 1, buf_v, &len_i) ) {
+    fprintf(stderr, "chop: failed to write last event\r\n");
+    u3_king_bail();
+  }
+
+  // backup the original database file
+  c3_c dat_c[8193], bak_c[8193];
+  snprintf(dat_c, sizeof(dat_c), "%s/data.mdb", log_c);
+  // "data_<first>-<last>.mdb.bak"
+  snprintf(bak_c, sizeof(bak_c), "%s/data_%" PRIu64 "-%" PRIu64 ".mdb.bak", cho_c, fir_d, las_d);
+  c3_rename(dat_c, bak_c);
+  if ( 0 != errno ) {
+    fprintf(stderr, "chop: failed to backup original database file\r\n");
+    u3_king_bail();
+  }
+
+  // rename new database file to be official
+  c3_c new_c[8193];
+  snprintf(new_c, sizeof(new_c), "%s/data.mdb", cho_c);
+  c3_rename(new_c, dat_c);
+  if ( 0 != errno ) {
+    fprintf(stderr, "chop: failed to rename new database file\r\n");
+    u3_king_bail();
+  }
+
+  // cleanup
+  u3_disk_exit(old_u);
+  u3_lmdb_exit(new_u);
+  u3m_stop();
+
+  // success
+  fprintf(stderr, "chop: event log truncation complete\r\n");
+  fprintf(stderr, "chop: event log backup written to %s\r\n", bak_c);
+  fprintf(stderr, "chop: WARNING: ENSURE YOU CAN RESTART YOUR SHIP "
+                  "BEFORE DELETING YOUR EVENT LOG BACKUP FILE!\r\n");
+  fprintf(stderr, "chop: if you can't restart your ship, restore your "
+                  "event log backup file with the following command:\r\n"); 
+  fprintf(stderr, "chop: mv %s %s\r\n", bak_c, dat_c);
+  fprintf(stderr, "chop: then restart your ship\r\n");
```

Comment:
These printfs can be combined, and should not all be prefixed like this. Subsequent output lines should be indented for vertical alignment, and wrapped to stay within 80 columns.